Update dependencies

k-yomo opened the pull request:
- Update Go version to 1.19
  - Some of the terraform related libraries require Go 1.18+ 
  (e.g.  https://github.com/hashicorp/terraform-plugin-sdk/tree/main#go-compatibility 
- Update terraform-plugin-sdk version to `v2.22.0`

elasticmachine commented:
Since this is a community submitted pull request, a Jenkins build has not been kicked off automatically. Can an Elastic organization member please verify the contents of this patch and then kick off a build manually?

tobio commented:
jenkins test this

k-yomo commented:
Lint was failing with false positive case.  I updated the golangci-lint version and refactored to pass the lint.
```
20:47:27  internal/clients/api_client.go:48:11: undeclared name: `elasticsearch` (typecheck)
20:47:27  	es      *elasticsearch.Client
```

@tobio 
Could you run the tests again?🙏

tobio commented:
jenkins test this
